ci_tutorial/products_pipeline/full_pipeline.rst

```diff
@@ -0,0 +1,303 @@
+
+.. code-block:: bash
+
+    $ conan lock create --requires=game/1.0 --lockfile-out=conan.lock
+    $ conan lock create --requires=game/1.0 -s build_type=Debug 
+      --lockfile=conan.lock --lockfile-out=conan.lock
+    $ conan lock create --requires=mapviewer/1.0 --lockfile=conan.lock 
+      --lockfile-out=conan.lock
+    $ conan lock create --requires=mapviewer/1.0 -s build_type=Debug 
+      --lockfile=conan.lock --lockfile-out=conan.lock
+
+
+.. note::
+
+    Recall that the ``conan.lock`` arguments are mostly optional, as that is the default lockfile name.
+    The first command can be typed as ``conan lock create --requires=game/1.0``. Also, all commands, including
+    ``conan install``, if they find a existing ``conan.lock`` file they will use it automatically, without an
+    explicit ``--lockfile=conan.lock``. The commands in this tutorial are shown explicitly complete for
+    completeness and didactical reasons.
+
+
+Then, we can compute the build order for each product and configuration. These commands are identical to the ones in the
+previous section, with the only difference of adding a ``--lockfile=conan.lock`` argument:
+
+
+.. code-block:: bash
+
+    $ conan graph build-order --requires=game/1.0 --lockfile=conan.lock 
+      --build=missing --order-by=recipe --format=json > game_release.json
+    $ conan graph build-order --requires=game/1.0 --lockfile=conan.lock 
+      --build=missing -s build_type=Debug --order-by=recipe --format=json > game_debug.json
+    $ conan graph build-order --requires=mapviewer/1.0 --lockfile=conan.lock 
+      --build=missing --order-by=recipe --format=json > mapviewer_release.json
+    $ conan graph build-order --requires=mapviewer/1.0 --lockfile=conan.lock 
+      --build=missing -s build_type=Debug --order-by=recipe --format=json > mapviewer_debug.json
+
+Likewise the ``build-order-merge`` command will be identical to the previous one.
+In this case, as this command doesn't really compute a dependency graph, a ``conan.lock`` argument is not necessary,
+dependencies are not being resolved:
+
+
+.. code-block:: bash
+
+    $ conan graph build-order-merge 
+      --file=game_release.json --file=game_debug.json 
+      --file=mapviewer_release.json --file=mapviewer_debug.json 
+      --reduce --format=json > build_order.json
+
+
+    
+So far, this process has been almost identical to the previous section one, just with the difference of capturing and using a lockfile.
+Now, we will explain the "core" of the ``products`` pipeline: iterating the build-order and distributing the build, gathering the 
+resulting built packages.
+
+This would be some Python code that performs the iteration sequentially (a real CI system would distribute the builds to different agents in parallel):
```

Comment:
```suggestion
This would be an example of some Python code that performs the iteration sequentially (a real CI system would distribute the builds to different agents in parallel):
```